verilog/formatting/tree_unwrapper.cc

```diff
@@ -1288,6 +1307,36 @@ void TreeUnwrapper::ReshapeTokenPartitions(
       break;
     }
 
+    case NodeEnum::kStatement: {
+      // Handle function calls: object.foo(arg);
+      auto* object_leaf = recent_partition->LeftmostDescendant();
+      if (object_leaf) {
+        const auto ftokens = object_leaf->Value().TokensRange();
+        if (ftokens.size() == 1) { // Only one token, check if we should merge it
+          auto& method_leaf = *ABSL_DIE_IF_NULL(object_leaf->NextLeaf());
+          if (PartitionStartsWithDot(method_leaf)) {
+            verible::MergeLeafIntoNextLeaf(object_leaf);
+          }
+        }
+      }
```

Comment:
This section might be a better fit for `kMacroCallExtension` and `kHierarchyExtension`, rather than `kStatement`.  Try to keep the transformation as local as possible.

Moving the semicolon at the `kStatement` level should be ok though.